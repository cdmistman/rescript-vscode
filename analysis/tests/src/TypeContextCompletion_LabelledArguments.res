type someVariant = One | Two | Three | Four | Five(int) | Six(option<string>, int)

type otherVariant = [#one | #two | #three | #four | #five(int) | #six(option<int>, int)]

let thisIsAValue = Two

let someVariantToString = (
  ~someVariant,
  ~anotherThing: TypeDefinition.variant,
  ~thirdThing: otherVariant,
) => {
  ignore(anotherThing)
  ignore(thirdThing)
  switch someVariant {
  | One => "One"
  | Two => "Two"
  | Three => "Three"
  | Four => "Four"
  | _ => "-"
  }
}

// let x = someVariantToString(~someVaria
//                                       ^com

// let x = someVariantToString(~someVariant=
//                                          ^com

// let x = someVariantToString(~someVariant=
//                                           ^com

// let x = someVariantToString(~someVariant=T
//                                           ^com

// let x = someVariantToString(~someVariant=t
//                                           ^com

// let x = someVariantToString(~someVariant=TypeDefinition.
//                                                         ^com

// let x = someVariantToString(~anotherThing=
//                                           ^com

// let x = someVariantToString(~thirdThing=
//                                         ^com

// let x = someVariantToString(~thirdThing=
//                                          ^com

// let x = someVariantToString(~thirdThing=#t
//                                           ^com

// let x = someVariantToString(~thirdThing=#T
//                                           ^com
