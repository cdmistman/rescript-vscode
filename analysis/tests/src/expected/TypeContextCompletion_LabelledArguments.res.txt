Complete src/TypeContextCompletion_LabelledArguments.res 22:41
posCursor:[22:41] posNoWhite:[22:40] Found expr:[22:11->22:41]
Pexp_apply ...[22:11->22:30] (~someVaria22:32->22:41=...[22:32->22:41])
Completable: CnamedArg(Value[someVariantToString], someVaria, [someVaria])
Found type for function (
  ~someVariant: someVariant,
  ~anotherThing: TypeDefinition.variant,
  ~thirdThing: otherVariant,
) => string
[{
    "label": "someVariant",
    "kind": 4,
    "tags": [],
    "detail": "someVariant",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 25:44
posCursor:[25:44] posNoWhite:[25:43] Found expr:[25:11->25:44]
Pexp_apply ...[25:11->25:30] (~someVariant25:32->25:43=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(NamedArg(someVariant, Value[someVariantToString])=)
Found variant type someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>, int)\n\n",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 28:45
posCursor:[28:45] posNoWhite:[28:43] Found expr:[28:11->28:44]
Pexp_apply ...[28:11->28:30] (~someVariant28:32->28:43=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(NamedArg(someVariant, Value[someVariantToString])=)
Found variant type someVariant
[{
    "label": "One",
    "kind": 4,
    "tags": [],
    "detail": "One\n\n",
    "documentation": null
  }, {
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "Four",
    "kind": 4,
    "tags": [],
    "detail": "Four\n\n",
    "documentation": null
  }, {
    "label": "Five(_)",
    "kind": 4,
    "tags": [],
    "detail": "Five(int)\n\n",
    "documentation": null
  }, {
    "label": "Six(_)",
    "kind": 4,
    "tags": [],
    "detail": "Six(option<string>, int)\n\n",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 31:45
posCursor:[31:45] posNoWhite:[31:44] Found expr:[31:11->31:45]
Pexp_apply ...[31:11->31:30] (~someVariant31:32->31:43=...[31:44->31:45])
found typed context 
Completable: CtypedContext(NamedArg(someVariant, Value[someVariantToString])=T)
Found variant type someVariant
[{
    "label": "Two",
    "kind": 4,
    "tags": [],
    "detail": "Two\n\n",
    "documentation": null
  }, {
    "label": "Three",
    "kind": 4,
    "tags": [],
    "detail": "Three\n\n",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 34:45
posCursor:[34:45] posNoWhite:[34:44] Found expr:[34:11->34:45]
Pexp_apply ...[34:11->34:30] (~someVariant34:32->34:43=...[34:44->34:45])
found typed context 
Completable: CtypedContext(NamedArg(someVariant, Value[someVariantToString])=t)
Found variant type someVariant
[{
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 37:59
posCursor:[37:59] posNoWhite:[37:58] Found expr:[37:11->37:59]
Pexp_apply ...[37:11->37:30] (~someVariant37:32->37:43=...[37:44->37:59])
found typed context 
Completable: CtypedContext(NamedArg(someVariant, Value[someVariantToString])=_)
Found variant type someVariant
[{
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 40:45
posCursor:[40:45] posNoWhite:[40:44] Found expr:[40:11->40:45]
Pexp_apply ...[40:11->40:30] (~anotherThing40:32->40:44=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(NamedArg(anotherThing, Value[someVariantToString])=)
Found variant type TypeDefinition.variant
[{
    "label": "Foo",
    "kind": 4,
    "tags": [],
    "detail": "Foo\n\n",
    "documentation": null
  }, {
    "label": "Bar",
    "kind": 4,
    "tags": [],
    "detail": "Bar\n\n",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 43:43
posCursor:[43:43] posNoWhite:[43:42] Found expr:[43:11->43:43]
Pexp_apply ...[43:11->43:30] (~thirdThing43:32->43:42=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(NamedArg(thirdThing, Value[someVariantToString])=)
Found polyvariant type otherVariant
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one",
    "documentation": null
  }, {
    "label": "#six(_)",
    "kind": 4,
    "tags": [],
    "detail": "#six(option<int>, int)",
    "documentation": null
  }, {
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }, {
    "label": "#five(_)",
    "kind": 4,
    "tags": [],
    "detail": "#five(int)",
    "documentation": null
  }, {
    "label": "#four",
    "kind": 4,
    "tags": [],
    "detail": "#four",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 46:44
posCursor:[46:44] posNoWhite:[46:42] Found expr:[46:11->46:43]
Pexp_apply ...[46:11->46:30] (~thirdThing46:32->46:42=...__ghost__[0:-1->0:-1])
found typed context 
Completable: CtypedContext(NamedArg(thirdThing, Value[someVariantToString])=)
Found polyvariant type otherVariant
[{
    "label": "#one",
    "kind": 4,
    "tags": [],
    "detail": "#one",
    "documentation": null
  }, {
    "label": "#six(_)",
    "kind": 4,
    "tags": [],
    "detail": "#six(option<int>, int)",
    "documentation": null
  }, {
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }, {
    "label": "#five(_)",
    "kind": 4,
    "tags": [],
    "detail": "#five(int)",
    "documentation": null
  }, {
    "label": "#four",
    "kind": 4,
    "tags": [],
    "detail": "#four",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 49:45
posCursor:[49:45] posNoWhite:[49:44] Found expr:[49:11->49:45]
Pexp_apply ...[49:11->49:30] (~thirdThing49:32->49:42=...[49:43->49:45])
found typed context 
Completable: CtypedContext(NamedArg(thirdThing, Value[someVariantToString])=t)
Found polyvariant type otherVariant
[{
    "label": "#three",
    "kind": 4,
    "tags": [],
    "detail": "#three",
    "documentation": null
  }, {
    "label": "#two",
    "kind": 4,
    "tags": [],
    "detail": "#two",
    "documentation": null
  }, {
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

Complete src/TypeContextCompletion_LabelledArguments.res 52:45
posCursor:[52:45] posNoWhite:[52:44] Found expr:[52:11->52:45]
Pexp_apply ...[52:11->52:30] (~thirdThing52:32->52:42=...[52:43->52:45])
found typed context 
Completable: CtypedContext(NamedArg(thirdThing, Value[someVariantToString])=T)
Found polyvariant type otherVariant
[{
    "label": "someVariantToString",
    "kind": 12,
    "tags": [],
    "detail": "(\n  ~someVariant: someVariant,\n  ~anotherThing: TypeDefinition.variant,\n  ~thirdThing: otherVariant,\n) => string",
    "documentation": null
  }]

